### PR TITLE
AEA-2509 Handle post-dated prescriptions in test-pack

### PR DESCRIPTION
**Pull request by booshibooshi**

### CODE REVIEW

File: tool/site/client/src/components/send/resultSummaries.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const ResultSummaries: React.FC<ResultSummariesProps> = ({sendResult}) =>
       <Label isPageHeading>Send Results</Label>
       <ButtonList>
         <Button onClick={() => copyPrescriptionIds(sendResult)}>Copy Prescription IDs</Button>
-        {sendResult.results.every(s => s.success !== "unknown")
+        {sendResult.results.every(s => s.success !== "unknown" && !s.success)
```
Comment (dan-jones-iw) [inline tool/site/client/src/components/send/resultSummaries.tsx:20]: does this new check invalidate the old one?

File: tool/site/client/src/fhir/helpers.ts [hunk 2]
```diff
@@ -114,12 +114,15 @@ export function updateBundleIds(bundle: fhir.Bundle): void {
 
 export function updateValidityPeriod(bundle: fhir.Bundle): void {
   const start = convertMomentToISODate(moment.utc())
-  const end = convertMomentToISODate(moment.utc().add(1, "month"))
   getMedicationRequestResources(bundle).forEach(medicationRequest => {
     const validityPeriod = medicationRequest.dispenseRequest?.validityPeriod
     if (validityPeriod) {
-      validityPeriod.start = start
-      validityPeriod.end = end
+      const oldStart = moment.utc(medicationRequest.dispenseRequest.validityPeriod.start)
```
Comment (dan-jones-iw) [inline tool/site/client/src/fhir/helpers.ts:120]: `const oldStart = moment.utc(validityPeriod.start)`
already have a variable for the above

File: tool/site/client/tests/pages/__snapshots__/sendPage.spec.tsx.snap [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 exports[`Displays confirmation page if multiple prescriptions are sent successfully 1`] = `
 "<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Send Results</label></h1>
-<div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy Prescription IDs</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"baseUrl/download/exception-report\\">Download Exception Report</a></div>
+<div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy Prescription IDs</button></div>
```
Comment (dan-jones-iw) [inline tool/site/client/tests/pages/__snapshots__/sendPage.spec.tsx.snap:5]: wheres the exception report gone?

File: tool/site/client/src/components/send/resultSummaries.tsx (hunk 1 above)
Comment (booshibooshi) [reply tool/site/client/src/components/send/resultSummaries.tsx:20]: no as the requests are batched deltas so the success status of unknown is what they all start as and overtime they all switch to true|false, if any are false we want to show the exception report

File: tool/site/client/tests/pages/__snapshots__/sendPage.spec.tsx.snap (hunk 3 above)
Comment (booshibooshi) [reply tool/site/client/tests/pages/__snapshots__/sendPage.spec.tsx.snap:5]: all prescriptions are succesful so the report is empty and we do not want to show it

File: tool/site/client/src/components/send/resultSummaries.tsx (hunk 1 above)
Comment (dan-jones-iw) [reply tool/site/client/src/components/send/resultSummaries.tsx:20]: perfect mate makes sense

Comment (booshibooshi) [reply tool/site/client/src/components/send/resultSummaries.tsx:20]: just the success for the moment, can pick up a negative test-case in future pr

**Review by dan-jones-iw — APPROVED** [review]

**Pull request merged**